src/carts/src/main/java/mushop/carts/controllers/CartsController.java

```diff
@@ -0,0 +1,163 @@
+import mushop.carts.entitites.Item;
+import mushop.carts.repositories.CartRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+@Controller("/carts")
+@ExecuteOn(TaskExecutors.IO)
+public class CartsController {
+    public static final Logger LOG = LoggerFactory.getLogger(CartsController.class);
+
+    private final CartRepository cartRepository;
+    private final MeterRegistry meterRegistry;
+
+    public CartsController(CartRepository cartRepository, MeterRegistry meterRegistry) {
+        this.cartRepository = cartRepository;
+        this.meterRegistry = meterRegistry;
+    }
+
+    @Get("/{cartId}")
+    public Cart getCart(String cartId) {
+        Cart cart = cartRepository.getById(cartId);
+        if (cart == null) {
+            throw new HttpStatusException(HttpStatus.NOT_FOUND, "Cart with id " + cartId + " not found");
+        }
+        return cart;
+    }
+
+    @Get("/{cartId}/items")
+    public List<Item> getCartItems(String cartId) {
+        Cart cart = cartRepository.getById(cartId);
+        if (cart == null) {
+            throw new HttpStatusException(HttpStatus.NOT_FOUND, "Cart with id " + cartId + " not found");
+        }
+        return cart.getItems();
+    }
+
+    @Delete("/{cartId}")
+    public Cart deleteCart(String cartId) {
+        Cart cart = cartRepository.getById(cartId);
+        if (cart == null) {
+            throw new HttpStatusException(HttpStatus.NOT_FOUND, "Cart with id " + cartId + " not found");
+        }
+
+        if (cartRepository.deleteCart(cartId)) {
+            meterRegistry.counter("carts.deleted").increment();
+            LOG.info("Cart deleted: {}", cart);
+            return cart;
+        } else {
+            throw new HttpStatusException(HttpStatus.NOT_FOUND, "Failed to delete cart " + cartId);
+        }
+    }
+
+    @Delete("/{cartId}/items/{itemId}")
+    public Cart deleteCartItem(String cartId, String itemId) {
+        Cart cart = cartRepository.getById(cartId);
+        if (cart == null) {
+            throw new HttpStatusException(HttpStatus.NOT_FOUND, "Cart with id " + cartId + " not found");
+        }
+
+        if (!cart.removeItem(itemId)) {
+            throw new HttpStatusException(HttpStatus.NOT_FOUND, "Cart item with id " + itemId + " not found " + cart);
+        }
+
+        meterRegistry.timer("carts.updated.timer").record(() -> {
+            cartRepository.save(cart);
+        });
+        meterRegistry.counter("carts.updated.counter").increment();
+        LOG.info("Item deleted: {}", cart);
+        return cart;
+    }
+
+    @Post("/{cartId}")
+    public HttpResponse<Cart> postCart(@PathVariable String cartId, @Body Cart newCart) {
+
+        Cart cart = cartRepository.getById(cartId);
+        if (cart == null) {
+            newCart.setId(cartId);
+            meterRegistry.timer("carts.created.timer").record(() -> {
+                cartRepository.save(newCart);
+            });
```

Comment:
This direct calls to `timer` can be replaced by adding `@Timed` to the methods you want to time